src/app.rs

```diff
@@ -489,7 +705,32 @@ impl<'a, 'b, T: State + 'a> ApplicationBuilder<'a, 'b, T> {
         self
     }
 
-    /// Register new context within the loader
+    /// Register a new asset type with the Application. All required components
+    /// related to the storage of this asset type will be registered. Since
+    /// Amethyst uses AssetFutures to allow for async content loading, Amethyst
+    /// needs to have a system that translates AssetFutures into Components as
+    /// they resolve. Amethyst includes a system to avoid the user having to
```

Comment:
Technically, I think it's included in Specs and not directly in Amethyst?

Reply:
Ok, ill clarify.